Drivers/Periphs/Inc/SDCard.h

```diff
@@ -1,21 +1,42 @@
+/**
+ * @file SDCard.h
+ * @brief SD Card API
+ * 
+ * @copyright Copyright (c) 2022 UT Longhorn Racing Solar
+ * 
+ */
+
 #ifndef SDCARD_H
 #define SDCARD_H
 
-#include <stdio.h>
-#include <string.h>
-#include <stdarg.h> //for va_list var arg functions
+#include "FreeRTOS.h"
+#include "queue.h"
 #include "fatfs.h"
 #include "CANBus.h"
 #include "IMU.h"
 #include "GPS.h"
 
+// writes will be performed in chunks less than/approximately this size
+#define SDCARD_WRITE_BUFSIZE        512
+#define SDCARD_SYNC_PERIOD          50      // sync period (milliseconds)
+#define SDCARD_MAX_MSGSIZE          128     // set this to equal or greater than the max message size 
```

Comment:
Feel like this would be a neat way of doing it:
`#define SIZEOF(x)                          (char *)(&x+1) - (char *)(&x)`
`#define SDCARD_MAX_MSGSIZE  SIZEOF(SDCardID_t)`

Reply:
That won't compile, but something like that...